messagebus/src/main/java/com/yahoo/messagebus/network/rpc/RPCTargetPool.java

```diff
@@ -95,17 +95,25 @@ public RPCTarget getTarget(Supervisor orb, RPCServiceAddress address) {
                 if (target != null) {
                     return target;
                 }
-                entry.close();
-                targets.remove(key);
+                dropTarget(entry, key);
             }
-            RPCTarget [] tmpTargets = new RPCTarget[numTargetsPerSpec];
-            for (int i=0; i < tmpTargets.length; i++) {
-                tmpTargets[i] = new RPCTarget(spec, orb);
-            }
-            entry = new Entry(tmpTargets, now);
-            targets.put(key, entry);
-            return entry.getTarget(now);
+            return createAndAddTarget(orb, spec, key, now);
+        }
+    }
+
+    private void dropTarget(Entry entry, String key) {
+        entry.close();
+        targets.remove(key);
+    }
+
+    private RPCTarget createAndAddTarget(Supervisor orb, Spec spec, String key, long now) {
```

Comment:
Technically should be `createAndAddTargets` in plural? Unless we want to hide this detail away

Reply:
Since there is no s in dropTarget I think it is symmetric as is.